src/main/scala/chisel3/util/TransitName.scala

```diff
@@ -2,25 +2,57 @@
 
 package chisel3.util
 
-import chisel3._
-import internal.HasId
+import chisel3.internal.HasId
 
-/**
- * The purpose of TransitName is to allow a library to 'move' a name
- * call to a more appropriate place.
- * For example, a library factory function may create a module and return
- * the io. The only user-exposed field is that given IO, which can't use
- * any name supplied by the user. This can add a hook so that the supplied
- * name then names the Module.
- * See Queue companion object for working example
- */
+/** The purpose of [[TransitName]] is to improve the naming of some object created in a different scope by "transiting"
+  * the name from the outer scope to the inner scope.
+  *
+  * Consider the example below. This shows three ways of instantiating `MyModule` and returning the IO. Normally, the
+  * instance will be named `MyModule`. However, it would be better if the instance was named using the name of the `val`
+  * that user provides for the returned IO. [[TransitName]] can then be used to "transit" the name ''from'' the IO
+  * ''to'' the module:
+  *
+  * {{{
+  * /* The instance name will be named "MyModule" */
+  * val foo = Module(new MyModule).io
+  *
+  * /* The instance will be named "bar" */
+  * val bar = {
+  *   val x = Module(new MyModule)
+  *   TransitName(x.io, x)
+  * }
+  *
+  * /* The instance will be named "baz_generated" */
+  * val baz = {
+  *   val x = Module(new MyModule)
+  *   TransitName.withSuffix("_generated")(x.io, x)
+  * }
+  * }}}
+  *
+  * [[TransitName]] help library writers following the [[https://en.wikipedia.org/wiki/Factory_method_pattern Factory
+  * Method Pattern]] where modules may be instantiated inside an enclosing scope. For an example of this, see the use of
+  * [[TransitName]] in the [[Queue$ Queue]] factory.
+  */
 object TransitName {
+
+  /** Transit a name from one type to another
+    * @param from the thing with a "good" name
+    * @param to the thing to get the "good" name
+    */
```

Comment:
Missing `@return`? (ditto below)

Reply:
Fixed in rebase. (Good catch... this should help clarify some confusion about how to use this.)